File: src/frontend/src/app/components/atendimento/tabelaregra/model/tabela-regra-dto.ts
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class TabelaRegraDTO {
  @Expose() id?: string;
  @Expose() convenioId?: string;
  @Expose() convenioNome?: string;
  @Expose() convenioCategoriaId?: string;
  @Expose() convenioCategoriaNome?: string;
  @Expose() tabelaId?: string;
  @Expose() tabelaNome?: string;
  @Expose() createdAt?: string;
  @Expose() updatedAt?: string;
  @Expose() createdBy?: string;
  @Expose() updatedBy?: string;
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/tabelaregra/dto/TabelaRegraDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.tabelaregra.dto;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class TabelaRegraDTO implements DTO {
    private UUID id;
    private UUID convenioId;
    private String convenioNome;
    private UUID convenioCategoriaId;
    private String convenioCategoriaNome;
    private UUID tabelaId;
    private String tabelaNome;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
